protocols/imap/src/main/java/org/apache/james/imap/processor/base/UidMsnConverter.java

```diff
@@ -46,13 +42,12 @@ public UidMsnConverter(Iterator<MessageUid> iterator) {
     }
 
     public synchronized Optional<Integer> getMsn(MessageUid uid) {
-        try {
-            int position = Ordering.explicit(uids).binarySearch(uids, uid);
-            return Optional.of(position + 1);
-        } catch (Exception e) {
-            LOGGER.trace("Error while converting uid to msn, uid not found: " + uid.asLong());
+        Collections.sort(uids);
```

Comment:
Do we really need to sort every time we get a MSN ?

I mean sort is a costly operation. I ensured to sort only on writes, if needed (if the new UID is not the biggest).